From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TinxLD <Ib0ekNtyFT93pp0Vi1ul>
Date: Tue, 4 Apr 2023 22:19:58 +0200
Subject: [PATCH] delme


diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index d3d0894b4a99005bea55e57f344437666e585dea..e79a451b0a76568a8bbdd85ef627f9188cca6742 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -25,7 +25,7 @@ public class HelpYamlReader {
     public HelpYamlReader(Server server) {
         this.server = server;
 
-        File helpYamlFile = new File("configa/help.yml");
+        File helpYamlFile = new File("config/help.yml");
         YamlConfiguration defaultConfig = YamlConfiguration.loadConfiguration(new InputStreamReader(getClass().getClassLoader().getResourceAsStream("configurations/help.yml"), Charsets.UTF_8));
 
         try {
